pkg/types/resource/security_insight_trait.go

```diff
@@ -0,0 +1,271 @@
+package resource
+
+import (
+	"fmt"
+	"time"
+
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	"github.com/conductorone/baton-sdk/pkg/annotations"
+)
+
+// SecurityInsightTraitOption is a functional option for configuring a SecurityInsightTrait.
+type SecurityInsightTraitOption func(*v2.SecurityInsightTrait) error
+
+// WithInsightValue sets the value of the security insight.
+func WithInsightValue(value string) SecurityInsightTraitOption {
+	return func(t *v2.SecurityInsightTrait) error {
+		t.SetValue(value)
+		return nil
+	}
+}
+
+// WithInsightObservedAt sets the observation timestamp for the insight.
+func WithInsightObservedAt(observedAt time.Time) SecurityInsightTraitOption {
+	return func(t *v2.SecurityInsightTrait) error {
+		t.SetObservedAt(timestamppb.New(observedAt))
+		return nil
+	}
+}
+
+// WithInsightContext sets additional context for the insight (deep links, remediation, etc.).
+func WithInsightContext(ctx map[string]interface{}) SecurityInsightTraitOption {
```

Comment:
it would be nice to normalize the "external link" like we do on other traits i think? so we know we can link to it?